digdag-standards/src/main/java/io/digdag/standards/operator/redshift/BaseRedshiftLoadOperator.java

```diff
@@ -0,0 +1,230 @@
+{
+    private static final String POLL_INTERVAL = "pollInterval";
+    private static final int INITIAL_POLL_INTERVAL = 1;
+    private static final int MAX_POLL_INTERVAL = 1200;
+
+    private static final String QUERY_ID = "queryId";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @VisibleForTesting
+    BaseRedshiftLoadOperator(OperatorContext context, TemplateEngine templateEngine)
+    {
+        super(context, templateEngine);
+    }
+
+    /* TODO: This method name should be connectionConfig() or something? */
+    @Override
+    protected RedshiftConnectionConfig configure(SecretProvider secrets, Config params)
+    {
+        return RedshiftConnectionConfig.configure(secrets, params);
+    }
+
+    /* TODO: This method should be in XxxxConnectionConfig ? */
+    @Override
+    protected RedshiftConnection connect(RedshiftConnectionConfig connectionConfig)
+    {
+        return RedshiftConnection.open(connectionConfig);
+    }
+
+    @Override
+    protected SecretProvider getSecretsForConnectionConfig()
+    {
+        return context.getSecrets().getSecrets("aws.redshift");
+    }
+
+    protected abstract List<SecretProvider> additionalSecretProvidersForCredentials(SecretProvider awsSecrets);
+
+    protected AWSCredentials createBaseCredential(SecretProvider secretProvider)
+    {
+        SecretProvider awsSecrets = secretProvider.getSecrets("aws");
+        SecretProvider redshiftSecrets = awsSecrets.getSecrets("redshift");
+        List<SecretProvider> secretProviders = additionalSecretProvidersForCredentials(awsSecrets);
+
+        String keyOfAccess = "access-key-id";
+        String accessKeyId =
+                secretProviders.stream()
+                        .map(sp -> sp.getSecretOptional(keyOfAccess))
+                        .filter(Optional::isPresent)
+                        .findFirst()
+                        .orElse(Optional.absent())
+                        .or(redshiftSecrets.getSecretOptional(keyOfAccess))
+                        .or(() -> awsSecrets.getSecret(keyOfAccess));
+
+        String keyOfSecret = "secret-access-key";
+        String secretAccessKey =
+                secretProviders.stream()
+                        .map(sp -> sp.getSecretOptional(keyOfSecret))
+                        .filter(Optional::isPresent)
+                        .findFirst()
+                        .orElse(Optional.absent())
+                        .or(redshiftSecrets.getSecretOptional(keyOfSecret))
+                        .or(() -> awsSecrets.getSecret(keyOfSecret));
+
+        return new BasicAWSCredentials(accessKeyId, secretAccessKey);
+    }
+
+    private AWSSessionCredentials createSessionCredentials(Config config, AWSCredentials baseCredential)
+    {
+        List<AcceptableUri> acceptableUris = buildAcceptableUriForSessionCredentials(config, baseCredential);
+
+        if (!config.get("temp_credentials", Boolean.class, true)) {
```

Comment:
There is a comment in `AWSSessionCredentialsFactory` class saying that `Maybe we'd better add an option command later like without_federated_token`. Isn't this `temp_credentials` option for the `without_federated_token`?